### ISSUE TITLE
VNSI: Requesting clients to load epg

### ISSUE THREAD

**Issue by AndreyPavlenko** [FernetMenta/vdr-plugin-vnsiserver#79]
The subj is printed to the syslog every 5 sec. Are there really any changes in the EPG every 5 sec?

**Comment by FernetMenta** [comment]
vnsi askes vdr if there is a change in epg database. if vdr says yes, vnsi interated through epg and looks for changes. if there is a change for a given channel you would see this log as well:
"Trigger EPG update for channel..."
if this log is missing vdr gives a false positive for some reason

**Comment by AndreyPavlenko** [comment]
When a client is connected, I can see many "VNSI-Debug: Trigger EPG update
for channel" messages followed by "VNSI: Requesting clients to load epg",
but after a few minutes I see only the "Requesting clients" messages
printed to the log every 5 seconds. Moreover, the message is duplicated:

Nov 10 22:15:15 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:15 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:20 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:20 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:26 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:26 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:31 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:31 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:36 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:36 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:41 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:41 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:46 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:46 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:51 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:51 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:15:56 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:15:56 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:01 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:01 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:06 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:06 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:11 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:11 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:16 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:16 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:21 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:21 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:26 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:26 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:31 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:31 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:36 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:36 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:41 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:41 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:46 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:46 htpc vdr: [26792] VNSI: Requesting clients to load epg
Nov 10 22:16:51 htpc vdr: [26794] VNSI: Requesting clients to load epg
Nov 10 22:16:51 htpc vdr: [26792] VNSI: Requesting clients to load epg

**Comment by FernetMenta** [comment]
looks normal. for some reason epg database is flagged as dirty by vdr or some other addon you may have loaded.

**Comment by AndreyPavlenko** [comment]
Ok, thanks. It seems does not impact on the functionality anyway.
Wouldn't it be better to change the log level to debug to avoid spamming to
the syslog?

**Comment by FernetMenta** [comment]
I would just hide an issue. Did you try with only vnsiserver and no other plugins?

**Comment by AndreyPavlenko** [comment]
No, I didn't. My plugins - dvbapi, restfulapi, streamdev, vnsiserver. I'll
try without restfulapi and streamdev.

On Fri, Nov 11, 2016 at 11:58 AM, Rainer Hochecker <notifications@github.com

**Comment by AndreyPavlenko** [comment]
No changes:

Nov 11 12:21:44 htpc vdr: [633] VNSI: Requesting clients to load epg
Nov 11 12:21:44 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:21:49 htpc vdr: [633] VNSI: Requesting clients to load epg
Nov 11 12:21:49 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:21:55 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:21:55 htpc vdr: [633] VNSI: Requesting clients to load epg
Nov 11 12:22:00 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:22:00 htpc vdr: [633] VNSI: Requesting clients to load epg
Nov 11 12:22:05 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:22:05 htpc vdr: [633] VNSI: Requesting clients to load epg
Nov 11 12:22:10 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:22:10 htpc vdr: [633] VNSI: Requesting clients to load epg
Nov 11 12:22:15 htpc vdr: [636] VNSI: Requesting clients to load epg
Nov 11 12:22:15 htpc vdr: [633] VNSI: Requesting clients to load epg

Why all the messages are duplicated? I've a single client connected.

**Comment by FernetMenta** [comment]
right, that does not look correct. I need to debug this.

**Comment by FernetMenta** [comment]
here is a fix: https://github.com/FernetMenta/vdr-plugin-vnsiserver/pull/80

**Comment by hoppel118** [comment]
Hello, 

I also had this "issue" tonight with a complete fresh system.

```
vdr (2.2.0/2.2.0) - The Video Disk Recorder
epgsearch (1.0.1.beta5) - search the EPG for repeats and more
vnsiserver (1.5.2) - VDR-Network-Streaming-Interface (VNSI) Server
epgsearchonly (0.0.1) - Direct access to epgsearch's search menu
conflictcheckonly (0.0.1) - Direct access to epgsearch's conflict check menu
live (0.3.0) - Live Interactive VDR Environment
quickepgsearch (0.0.1) - Quick search for broadcasts
```
All packages are from e-tobis repo, except vnsiserver. vnsiserver is compiled from latest github.

I reduced the loglevel from 2 to 1. Now I don't see the messages anymore.

Greetings Hoppel
